2025-04-18 12:51:48.259140: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-04-18 12:51:50.513848: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2025-04-18 12:51:50.513895: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2025-04-18 12:51:50.728653: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.1.3 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/aver/Machine_Learning/AML_MiniProject/src/data_preprocessing.py", line 21, in <module>
    import tensorflow as tf
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/__init__.py", line 37, in <module>
    from tensorflow.python.eager import context
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/eager/context.py", line 35, in <module>
    from tensorflow.python.client import pywrap_tf_session
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/client/pywrap_tf_session.py", line 19, in <module>
    from tensorflow.python.client._pywrap_tf_session import *
AttributeError: _ARRAY_API not found

A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.1.3 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.

Traceback (most recent call last):  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/aver/Machine_Learning/AML_MiniProject/src/data_preprocessing.py", line 21, in <module>
    import tensorflow as tf
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/__init__.py", line 42, in <module>
    from tensorflow.python import data
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/__init__.py", line 21, in <module>
    from tensorflow.python.data import experimental
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/__init__.py", line 96, in <module>
    from tensorflow.python.data.experimental import service
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/service/__init__.py", line 419, in <module>
    from tensorflow.python.data.experimental.ops.data_service_ops import distribute
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/data_service_ops.py", line 24, in <module>
    from tensorflow.python.data.experimental.ops import compression_ops
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/compression_ops.py", line 16, in <module>
    from tensorflow.python.data.util import structure
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/util/structure.py", line 23, in <module>
    from tensorflow.python.data.util import nest
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/util/nest.py", line 36, in <module>
    from tensorflow.python.framework import sparse_tensor as _sparse_tensor
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/sparse_tensor.py", line 24, in <module>
    from tensorflow.python.framework import constant_op
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/constant_op.py", line 25, in <module>
    from tensorflow.python.eager import execute
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/eager/execute.py", line 23, in <module>
    from tensorflow.python.framework import dtypes
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/dtypes.py", line 29, in <module>
    from tensorflow.python.lib.core import _pywrap_bfloat16
AttributeError: _ARRAY_API not found
ImportError: numpy.core._multiarray_umath failed to import
ImportError: numpy.core.umath failed to import
Traceback (most recent call last):
  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/aver/.local/share/uv/python/cpython-3.10.17-linux-x86_64-gnu/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/aver/Machine_Learning/AML_MiniProject/src/data_preprocessing.py", line 21, in <module>
    import tensorflow as tf
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/__init__.py", line 42, in <module>
    from tensorflow.python import data
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/__init__.py", line 21, in <module>
    from tensorflow.python.data import experimental
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/__init__.py", line 96, in <module>
    from tensorflow.python.data.experimental import service
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/service/__init__.py", line 419, in <module>
    from tensorflow.python.data.experimental.ops.data_service_ops import distribute
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/data_service_ops.py", line 24, in <module>
    from tensorflow.python.data.experimental.ops import compression_ops
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/experimental/ops/compression_ops.py", line 16, in <module>
    from tensorflow.python.data.util import structure
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/util/structure.py", line 23, in <module>
    from tensorflow.python.data.util import nest
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/data/util/nest.py", line 36, in <module>
    from tensorflow.python.framework import sparse_tensor as _sparse_tensor
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/sparse_tensor.py", line 24, in <module>
    from tensorflow.python.framework import constant_op
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/constant_op.py", line 25, in <module>
    from tensorflow.python.eager import execute
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/eager/execute.py", line 23, in <module>
    from tensorflow.python.framework import dtypes
  File "/home/aver/Machine_Learning/AML_MiniProject/.venv/lib/python3.10/site-packages/tensorflow/python/framework/dtypes.py", line 34, in <module>
    _np_bfloat16 = _pywrap_bfloat16.TF_bfloat16_type()
TypeError: Unable to convert function return value to a Python type! The signature was
	() -> handle
